### DISCUSSION TITLE
How to make file audio load quickly

### DISCUSSION THREAD

**Discussion by bugoverfl0w** [savonet/liquidsoap#1795] [Q&A]
Dear team,

I use liquidsoap + icecast to stream audio,
```
output.icecast(%mp3,
               host = "localhost", port = 8000,
               password = "hackme", mount = channel,
               main_stream
               )
```
In HTML page I use audio tag to load mp3 stream link.

Currently it takes about 6-8 seconds to load then play in web browser.

In VLC it starts in 2 seconds, and in mobile app, the same as VLC

I was researched and test some help on Google but nothing okay

How can I reduce this time on web browser?

Thanks,

**Comment by kobayashi90** [comment]
you should setup your icecast correctly. let me know if you still need help.

**Comment by bugoverfl0w** [comment]
> you should setup your icecast correctly. let me know if you still need help.

Thanks for your reply. Could you please give me more information?

**Comment by kobayashi90** [comment]
> 
> 
> > you should setup your icecast correctly. let me know if you still need help.
> 
> Thanks for your reply. Could you please give me more information?

replace these settings on your icecast.xml file which should be at /etc/icecast2/icecast.xml

```
<limits>
<clients>100</clients>
<sources>10</sources>
<queue-size>1048576</queue-size>
<threadpool>5</threadpool>
<client-timeout>20</client-timeout>
<header-timeout>15</header-timeout>
<source-timeout>60</source-timeout>
<burst-on-connect>0</burst-on-connect>
<burst-size>943718</burst-size>
</limits>
```

You can fix the clients & sources numbers to the amount of your liking.

**Comment by bugoverfl0w** [comment]
> > > you should setup your icecast correctly. let me know if you still need help.
> > 
> > 
> > Thanks for your reply. Could you please give me more information?
> 
> replace these settings on your icecast.xml file which should be at /etc/icecast2/icecast.xml
> 
> ```
> <limits>
> <clients>100</clients>
> <sources>10</sources>
> <queue-size>1048576</queue-size>
> <threadpool>5</threadpool>
> <client-timeout>20</client-timeout>
> <header-timeout>15</header-timeout>
> <source-timeout>60</source-timeout>
> <burst-on-connect>0</burst-on-connect>
> <burst-size>943718</burst-size>
> </limits>
> ```
> 
> You can fix the clients & sources numbers to the amount of your liking.

Thanks bro,

I know this one, this will make stream buffer, so it will load fastly when client first connect.

But it has delay time with two tracks. For example

when the on_track event called, the current song must play about 20-30s before the next song start. How can you fix this issue
